pkg/git/providers/git/git.go

```diff
@@ -0,0 +1,69 @@
+package git
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+)
+
+const (
+	GitPasswordTokenEnv       = "GIT_PASSWORD"
+	EksaGitPasswordTokenEnv   = "EKSA_GIT_PASSWORD"
+	EksaGitPrivateKeyTokenEnv = "EKSA_GIT_PRIVATE_KEY"
+)
+
+type Options struct {
+	RepositoryUrl string
+	User          string
+}
+
+func GetGitPasswordFromEnv() (string, error) {
+	password, err := GetTokenFromEnv(EksaGitPasswordTokenEnv)
+	if err != nil {
+		return "", err
+	}
+	if len(password) > 0 {
+		if err := os.Setenv(GitPasswordTokenEnv, password); err != nil {
+			return "", fmt.Errorf("unable to set %s: %v", GitPasswordTokenEnv, err)
+		}
+	}
+
+	return password, nil
+}
+
+func GetGitPrivateKeyFileFromEnv() (string, error) {
+	privateKeyFilePath, err := GetTokenFromEnv(EksaGitPrivateKeyTokenEnv)
+	if err != nil {
+		return "", err
+	}
+	privateKeyfileExistsAndIsNotEmpty := validateFilePath(privateKeyFilePath)
+	if !privateKeyfileExistsAndIsNotEmpty {
+		logger.V(4).Info("Private key file does not exist or is empty", "file path", privateKeyFilePath)
+		return "", fmt.Errorf("private key file does not exist at %s or is empty", privateKeyFilePath)
```

Comment:
This would fail during the flux execution
Should this be a pre flight validation of some sorts?